data_collection/scheduler.py

```diff
@@ -1,13 +1,38 @@
 import datetime
 
 import click
-import enabled_spiders
 from decouple import config
 from scrapinghub import ScrapinghubClient
+from sqlalchemy import create_engine, select, update
+from sqlalchemy.orm import sessionmaker
+
+from gazette.database.models import QueridoDiarioSpider
 
 YESTERDAY = datetime.date.today() - datetime.timedelta(days=1)
 
 
+def get_enabled_spiders(start_date=None, end_date=None):
+    """Return list of all currently enabled spiders within date period.
+
+    If start_date and/or end_date are provided, it will return only
+    the enabled spiders that are within the requested date period. Otherwise
+    it will execute a full crawl (no limited date period).
+    """
+    engine = create_engine(config("QUERIDODIARIO_DATABASE_URL"))
+    Session = sessionmaker(bind=engine)
+    session = Session()
+
+    stmt = select(QueridoDiarioSpider).where(QueridoDiarioSpider.enabled.is_(True))
+    if start_date is not None:
+        stmt = stmt.where(QueridoDiarioSpider.date_from <= start_date)
+    if end_date is not None:
+        stmt = stmt.where(QueridoDiarioSpider.date_to >= end_date)
```

Comment:
Essa verificação funciona caso o `.date_to` seja nulo?

Reply:
Se `date_to` for nulo, não irá retornar nenhum valor se um `end_date` for fornecido. Acho que faz sentido, já que se fornecemos `end_date` estamos esperando um período fechado específico.

Até poderíamos considerar o valor nulo como válido, mas acho que o uso será muito específico (nenhuma action fornece o `end_date` por padrão) e o código pode ficar um pouco mais confuso.

Reply:
Considerando o padrão para o campo `date_to` definido como `'null' if able to gather data in current day`, que tal:
```python
    if end_date is not None:
        stmt = stmt.where(
            or_(
                QueridoDiarioSpider.date_to == None,
                and_(
                    QueridoDiarioSpider.date_to != None,
                    QueridoDiarioSpider.date_to >= end_date
                )
            )
        )
```